storm-client/src/jvm/org/apache/storm/StormSubmitter.java

```diff
@@ -235,7 +237,14 @@ public static void submitTopologyAs(String name, Map<String, Object> topoConf, S
         conf.putAll(topoConf);
         topoConf.putAll(prepareZookeeperAuthentication(conf));
 
-        validateConfs(conf, topology, name);
+        validateConfs(conf);
+
+        try {
+            StormCommon.validateCycleFree(topology, name);
+        } catch (InvalidTopologyException ex) {
+            LOG.warn(ex.get_msg(), ex);
```

Comment:
Will `ex` already contains `ex.get_msg()`? If so, we will see repeating messages, which should be avoid.

Reply:
The constructor seems to imply that the message will not be in the Exception, since the proper super method is not called.

Reply:
Do you have an output example that I can take a look?

Reply:
InvalidTopologyException constructor does not call Exception.super(msg): https://github.com/apache/storm/blob/master/storm-client/src/jvm/org/apache/storm/generated/InvalidTopologyException.java#L111

hence, its msg variable is not available in the Exception.getMessage()

Reply:
Right. So the log will look like 
```
19:22:23.981 [main] WARN  o.a.s.StormSubmitter - Topology wc contains cycles in components "count,split"
org.apache.storm.generated.InvalidTopologyException: null
	at org.apache.storm.daemon.StormCommon.validateCycleFree(StormCommon.java:596) ~[storm-client-2.3.0-SNAPSHOT.jar:2.3.0-SNAPSHOT]
	at org.apache.storm.StormSubmitter.submitTopologyAs(StormSubmitter.java:244) [storm-client-2.3.0-SNAPSHOT.jar:2.3.0-SNAPSHOT]
	at org.apache.storm.StormSubmitter.submitTopology(StormSubmitter.java:214) [storm-client-2.3.0-SNAPSHOT.jar:2.3.0-SNAPSHOT]
	at org.apache.storm.StormSubmitter.submitTopology(StormSubmitter.java:177) [storm-client-2.3.0-SNAPSHOT.jar:2.3.0-SNAPSHOT]
	at org.apache.storm.topology.ConfigurableTopology.submit(ConfigurableTopology.java:119) [storm-client-2.3.0-SNAPSHOT.jar:2.3.0-SNAPSHOT]
	at org.apache.storm.starter.WordCountTopology.run(WordCountTopology.java:58) [storm-starter-2.3.0-SNAPSHOT.jar:2.3.0-SNAPSHOT]
	at org.apache.storm.topology.ConfigurableTopology.start(ConfigurableTopology.java:68) [storm-client-2.3.0-SNAPSHOT.jar:2.3.0-SNAPSHOT]
	at org.apache.storm.starter.WordCountTopology.main(WordCountTopology.java:36) [storm-starter-2.3.0-SNAPSHOT.jar:2.3.0-SNAPSHOT]
```


We can probably use [WrappedInvalidTopologyException](https://github.com/apache/storm/blob/master/storm-client/src/jvm/org/apache/storm/utils/WrappedInvalidTopologyException.java) inside `validateCycleFree` method so the stacktrace will make more sense. What do you think? 
It will be like 
```
org.apache.storm.utils.WrappedInvalidTopologyException: Topology wc2 contains cycles in components "count,split"
	at org.apache.storm.daemon.StormCommon.validateCycleFree(StormCommon.java:596) ~[storm-client-2.3.0-SNAPSHOT.jar:2.3.0-SNAPSHOT]
	.....
```

Reply:
That is better than a scary "null" in the stack trace.
Changed.